fairing/backends/backends.py

```diff
@@ -109,7 +146,11 @@ def get_builder(self, preprocessor, base_image, registry, needs_deps_installatio
         pod_spec_mutators.append(gcp.add_gcp_credentials_if_exists)
         # TODO (karthikv2k): Add cloud build as the deafult
         # once https://github.com/kubeflow/fairing/issues/145 is fixed
-        if fairing.utils.is_running_in_k8s():
+        if not needs_deps_installation:
```

Comment:
This logic for deciding which Builder to use would be good to factor into a separate function that returns an Enum or equivalent in python. that would make it easier to keep in sync at multiple invocations.

Reply:
I made the changes but let me do that as a separate PR to keep the changes to minimum.